lib/timers.js

```diff
@@ -580,13 +580,15 @@ function processImmediate() {
   while (immediate) {
     domain = immediate.domain;
 
-    if (!immediate._onImmediate)
-      continue;
```

Comment:
This doesn't seem correct. We should definitely always skip to the next here if there is no callback (`_onImmediate`).

Would it work if you moved https://github.com/nodejs/node/pull/9759/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR594 above this?